packages/server/src/main/java/software/uncharted/terarium/hmiserver/utils/ModelConfigurationLatexTable.java

```diff
@@ -0,0 +1,162 @@
+
+public class ModelConfigurationLatexTable {
+
+	public static String generateLatexTable(Model model, ModelConfiguration modelConfiguration) {
+		StringBuilder latex = new StringBuilder();
+
+		// Start LaTeX table
+		latex.append("\\documentclass{article}\n");
+		latex.append("\\usepackage{array, float, booktabs, subcaption}\n");
+		latex.append("\\begin{document}\n");
+		latex.append("\\begin{table}\n");
+
+		// Generate Initial Values Table
+		if (!modelConfiguration.getInitialSemanticList().isEmpty()) {
+			latex.append("\\begin{center}\n");
+			latex.append("\\caption{Initial expressions of the model state variables}\n");
+			latex.append("\\begin{tabular}{cccc>{\\small}m{8em}>{\\small}m{4em}}\n");
+			latex.append("\\toprule\n");
+			latex.append("ID & Name & Expression & Units & Description & Source \\\\\n");
+			latex.append("\\midrule\n");
+
+			for (InitialSemantic initial : modelConfiguration.getInitialSemanticList()) {
+				String id = escapeLatex(initial.getTarget());
+				String name = "";
+				String expression = escapeLatex(initial.getExpression());
+				String units = "";
+				String description = "";
+				String source = escapeLatex(initial.getSource());
+
+				// Get the values from the state
+				final State state = model
+					.getStates()
+					.stream()
+					.filter(s -> s.getId().equals(initial.getTarget()))
+					.findFirst()
+					.orElse(null);
+
+				if (state != null) {
+					name = escapeLatex(state.getName());
+					description = escapeLatex(state.getDescription());
+					units = escapeLatex(state.getUnits().getExpression());
+				}
+
+				// Add the row to the table
+				latex.append(
+					String.format("%s & %s & %s & %s & %s & %s \\\\\n", id, name, expression, units, description, source)
+				);
+				latex.append("\\hline\n");
+			}
+
+			// remove the last \\hline
+			latex.delete(latex.length() - 8, latex.length());
+			latex.append("\n");
+
+			latex.append("\\bottomrule\n");
+			latex.append("\\end{tabular}\n");
+			latex.append("\\end{center}\n");
+		}
+
+		// Generate Parameter Table
+		if (!modelConfiguration.getParameterSemanticList().isEmpty()) {
+			latex.append("\\hfill\n");
+			latex.append("\\begin{center}\n");
+			latex.append("\\caption{Model parameters}\n");
+			latex.append("\\begin{tabular}{cccc>{\\small}m{8em}>{\\small}m{4em}}\n");
+			latex.append("\\toprule\n");
+			latex.append("ID & Name & Value & Units & Description & Source \\\\\n");
+			latex.append("\\midrule\n");
+
+			for (ParameterSemantic param : modelConfiguration.getParameterSemanticList()) {
+				String id = escapeLatex(param.getReferenceId());
+				String name = "";
+				String value = "";
+				String units = "";
+				String description = "";
+				String source = escapeLatex(param.getSource());
+
+				// Get the values from the Parameter
+				final ModelParameter parameter = model
+					.getParameters()
+					.stream()
+					.filter(s -> s.getId().equals(param.getReferenceId()))
+					.findFirst()
+					.orElse(null);
```

Comment:
Consider caching states and parameters so this won't become a perf problem with larger models.

Something like
```
final Map<UUID, State> map = new HashMap<>();
for (final State state: model.getStates()) {
  map.put(state.getId(), state);
}

...

for (InitialSemantic initial : modelConfiguration.getInitialSemanticList()) {
  ...
  state = map.get(initial.getTarget());
 ...
}

```

Reply:
Fixed! Thanks @mwdchang